src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrBack = [];
+  let newState = {};
+
+  Object.assign(newState, state);
+
+  for (const act of actions) {
+    switch (act.type) {
+      case 'addProperties':
+        Object.assign(newState, act.extraData);
```

Comment:
Using ``Object.assign(newState, state)`` works, but it's better to use destructuring to create a copy of the object: